hack/make/agent.make

```diff
@@ -1,3 +1,5 @@
+##@ Agent
```

Comment:
@philbrookes fyi, adding these makes the help nicer, see :

```
$ make help

Usage:
  make <target>

General
  help             Display this help.

Development
  clean            Clean up temporary files.
  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
  generate         Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
  fmt              Run go fmt against code.
  vet              Run go vet against code.
  lint             Run golangci-lint against code.
  test             Run tests.
  local-setup      Setup multi cluster traffic controller locally using kind.
  local-cleanup    Cleanup resources created by local-setup
  build            Build all binaries.

Deployment
  install          Install CRDs into the K8s cluster specified in ~/.kube/config.
  uninstall        Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
  dev-tls          Generate dev tls webhook cert if necessary.

Agent
  build-agent      Build agent binary.
  docker-build-agent  Build docker image with the agent.
  docker-push-agent  Push docker image with the agent.
  deploy-agent     Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy-agent   Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.

Controller
  build-controller  Build controller binary.
  docker-build-controller  Build docker image with the controller.
  docker-push-controller  Push docker image with the controller.
  deploy-controller  Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy         Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.

Syncer
  build-syncer     Build syncer binary.
  docker-build-syncer  Build docker image with the syncer.
  docker-push-syncer  Push docker image with the syncer.
  deploy-syncer    Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy-syncer  Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.

Build Dependencies
  kustomize        Download kustomize locally if necessary. If wrong version is installed, it will be removed before downloading.
  controller-gen   Download controller-gen locally if necessary. If wrong version is installed, it will be overwritten.
  envtest          Download envtest-setup locally if necessary.
  kind             Download kind locally if necessary.
```